### PR TITLE
enable -march=native by default

### PR DESCRIPTION
**Pull request by stefantalpalaru**
plus some drive-by fixes and cosmetic changes

### CODE REVIEW

File: nimbus.nimble [hunk 1]
```diff
@@ -26,7 +27,7 @@ proc buildBinary(name: string, srcDir = "./", params = "", lang = "c") =
   var extra_params = params
   for i in 2..<paramCount():
     extra_params &= " " & paramStr(i)
-  exec "nim " & lang & " --out:build/" & name & " " & extra_params & " " & srcDir & name & ".nim"
+  exec "nim " & lang & " --out:build/" & name & " -d:release --import:libbacktrace " & extra_params & " " & srcDir & name & ".nim"
```
Comment (zah) [inline nimbus.nimble]: The parameters which always stay the same (such as `--import:libbacktrace`) should be preferably specified in cfg files.
This ensures that manual builds from the command line and in scripts such as `tests/simulation/start.sh` use them as well.

Comment (stefantalpalaru) [reply nimbus.nimble]: Tried it and it doesn't work. I documented that here: https://github.com/status-im/nimbus/blob/e9e33d3a56f240b7c31f3e093fac716defff4b7c/config.nims#L31

Comment (zah) [reply nimbus.nimble]: I see. Let's make sure that `tests/simulation/start.sh` uses the new flags as well then.

Comment (stefantalpalaru) [reply nimbus.nimble]: Yes, looking into that.

**Pull request merged**